staging/src/k8s.io/apimachinery/pkg/watch/watch.go

```diff
@@ -161,13 +163,16 @@ func (a *AggregatedWatcher) AddWatchInterface(watcher Interface, err error) {
 
 func (a *AggregatedWatcher) closeWatcher(watcher Interface, stopCh *bcast.Member) {
 	watcher.Stop()
+	a.wg.Done()
 	a.stopChGrp.Leave(stopCh)
 	a.removeWatcherAndError(watcher)
 
 	if !a.allowWatcherReset && a.stopChGrp.MemberCount() == 0 {
-		//klog.Infof("Close watcher %v caused aggregated channel %v closed", watcher, a.aggChan)
+		//klog.V(4).Infof("Close watcher %v caused aggregated channel %v closed", watcher, a.aggChan)
```

Comment:
My suggestion is to uncomment the logs. It might be good to collect information in case  there are other close channel issues.